common/src/app/common/types/container.cljc

```diff
@@ -211,20 +211,25 @@
     :else
     (get-instance-root objects (get objects (:parent-id shape)))))
 
-(defn inside-component-main?
-  "Check if the shape is a component main instance or is inside one."
+(defn find-component-main
+  "If the shape is a component main instance or is inside one, return that instance"
   [objects shape]
   (cond
     (or (nil? shape) (cfh/root? shape))
-    false
+    nil
     (nil? (:parent-id shape))  ; This occurs in the root of components v1
-    true
+    shape
     (ctk/main-instance? shape)
-    true
-    (ctk/instance-head? shape)
```

Comment:
This function is used from `in-any-component?`, below. And this one in turn is called from several places around the code. I think it's dangerous to change the behavior. Perhaps it's better to add a flag and leave the default behavior as it was before.